src/main/java/me/moodcat/database/BulkInsertData.java

```diff
@@ -0,0 +1,219 @@
+     * 
+     * @param username
+     *            the username of the artist.
+     * @param artistMap
+     *            the map with already persisted arists.
+     */
+    private Artist getOrPersistArtist(String username, HashMap<String, Artist> artistMap) {
+        final ArtistDAO artistDAO = artistDAOProvider.get();
+        Artist artist;
+        if (artistMap.containsKey(username)) {
+            artist = artistMap.get(username);
+        } else {
+            artist = new Artist();
+            artist.setName(username);
+            artistDAO.persist(artist);
+        }
+        return artist;
+    }
+
+    /**
+     * Insert a given amount of random generated rooms into the database. This is used for testing
+     * purposes.
+     * 
+     * @param numberOfRooms
+     *            the amount of rooms to generate.
+     */
+    public void insertRandomRooms(int numberOfRooms) {
+        final RoomDAO roomDAO = roomDAOProvider.get();
+        final SongDAO songDAO = songDAOProvider.get();
+        List<Song> songs = songDAO.listSongs();
+        Random random = new Random();
+        for (int i = 0; i < numberOfRooms; i++) {
+            Room room = new Room();
+            System.out.println(songs.size());
+            room.setSong(songs.get(random.nextInt(songs.size())));
+            room.setName("ROOM_STUB #" + i);
+            room.setTime(0);
+            room.setPosition(i);
+            room.setChatMessages(Collections.<ChatMessage> emptyList());
+            roomDAO.persist(room);
+        }
+    }
+
+    /**
+     * Helper method to create a {@link Song} given a {@link SoundCloudTrack} and an id,
+     * {@link Artist} and {@link VAVector}.
+     * 
+     * @param track
+     *            the given track.
+     * @param id
+     *            the given SoundCloud id.
+     * @param artist
+     *            the given {@link Artist}
+     * @param vector
+     *            the given {@link VAVector}
+     * @return the resulting song.
+     */
+    private Song songToTrack(SoundCloudTrack track, int id, Artist artist, VAVector vector) {
+        Song song = new Song();
+        song.setName(track.getTitle());
+        song.setSoundCloudId(id);
+        song.setDuration(track.getDuration());
+        song.setArtist(artist);
+        song.setArtworkUrl(track.getArtworkUrl());
+        song.setValenceArousal(vector);
+        return song;
+    }
+
+    /**
+     * Read and parse the file specified in SOUNDCLOUD_ID_FILE_PATH and return an int array.
+     * 
+     * @return the parsed int array.
+     * @throws IOException
+     *             when the file is not found or can not be read.
+     */
+    private int[] readSoundCloudIds() throws IOException {
+        String soundcloudIdString = new String(Files.readAllBytes(Paths
+                .get(SOUNDCLOUD_ID_FILE_PATH)));
+        String[] soundcloudIds = soundcloudIdString.split("\n");
+        int[] ids = new int[soundcloudIds.length];
```

Comment:
If we use maps here we can use map:

```java
List<String> ids = Lists.newArrayList(soundCloudIds);
return ids
    .map(Integer::parseInt)
    .collect(Collectors.asList());
```
Its however not much more efficient due to the collecting phase. It is however considerably more readable and has less statements / branches.